src/jvm/test/org/jetbrains/kotlinx/lincheck_test/gpmc/DaemonThreadPoolTest.kt

```diff
@@ -10,25 +10,147 @@
+            //                 abort background threads, when main is finished
+        }
+    }
+
+    @Test
+    fun testUselessThreadPoolWithMultipleCoroutines() {
+        runConcurrentTest(1000) {
+            val pool = Executors.newFixedThreadPool(5).asCoroutineDispatcher()
+            runBlocking(pool) {
+                val c1 = launch(pool) {}
+                val c2 = launch(pool) {}
+                val c3 = launch(pool) {}
+                val c4 = launch(pool) {}
+                val c5 = launch(pool) {}
+
+                c1.join()
+                c2.join()
+                c3.join()
+                c4.join()
+                c5.join()
+            }
+            // pool.close() -- no manual closing
+        }
+    }
+
+    @Test
+    fun testInfiniteBackgroundThreads() {
+        runConcurrentTest(1000) {
+            val atomic = AtomicInteger(0)
+            thread {
+                while (true) atomic.incrementAndGet()
+            }
+            val t2 = thread(start = false) {
+                while (true) atomic.decrementAndGet()
+            }
+            t2.isDaemon = true
+            t2.start()
+        }
+    }
+
+    @Test
+    fun testManyInfiniteBackgroundThreads() {
+        runConcurrentTest(1000) {
+            val atomic = AtomicInteger(0)
+            thread {
+                while (true) atomic.incrementAndGet()
+            }
+            thread {
+                while (true) atomic.incrementAndGet()
+            }
+            thread {
+                while (true) atomic.incrementAndGet()
+            }
+            thread {
+                while (true) atomic.incrementAndGet()
+            }
+        }
+    }
+
+    @Test(expected = LincheckAssertionError::class)
+    fun testBackgroundThreadsDeadlock() {
+        runConcurrentTest(1000) {
+            val sync1 = Any()
+            val sync2 = Any()
+            thread(start = false) {
+                synchronized(sync1) {
+                    synchronized(sync2) {
+                    }
+                }
+            }.start()
+
+            thread(start = false) {
+                synchronized(sync2) {
+                    synchronized(sync1) {
+                    }
+                }
+            }.start()
+        }
+    }
+
+    @Test(expected = LincheckAssertionError::class)
+    fun testBackgroundCoroutinesDeadlock() {
+        runConcurrentTest(1000) {
+            Executors.newFixedThreadPool(2).asCoroutineDispatcher().use { pool ->
+                val m = Mutex()
+                runBlocking(pool) {
+                    launch(pool) {
+                        m.lock()
+                    }
+                    launch(pool) {
+                        m.lock()
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testJavaThreadPoolFutures() {
+        runConcurrentTest(1000) {
+            val pool = Executors.newFixedThreadPool(3)
+            val f1 = pool.submit {}
```

Comment:
Also better to add some code inside tasks (like atomic integer increment).
In other tests as well.